### PR TITLE
SWC-5257: use fancy cell editor for entity id list and user id list. …

### PR DESCRIPTION
**Pull request by jay-hodgson**
… and use proper single cell editors in the new modal for these list types.

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/table/v2/results/cell/EditJSONListModal.java [hunk 1]
```diff
@@ -26,12 +26,14 @@
 public class EditJSONListModal implements EditJSONListModalView.Presenter{
 	public static final String SEE_THE_ERRORS_ABOVE = "See the error(s) above.";
 
-	private static final Map<ColumnType, ColumnType> LIST_TYPE_TO_NON_LIST = ImmutableMap.of(
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/table/v2/results/cell/EditJSONListModal.java:29]: apparently ImmutableMap.of() only supports up to 10 params (2, 4, 6, 8, or 10)!

File: src/main/java/org/sagebionetworks/web/client/widget/table/v2/results/cell/JSONListCellEditorView.java [hunk 2]
```diff
@@ -1,14 +1,10 @@
...
-import org.sagebionetworks.repo.model.table.ColumnModel;
-import org.sagebionetworks.repo.model.table.ColumnType;
-
 public interface JSONListCellEditorView extends CellEditorView {
 	public interface Presenter{
 		void onEditButtonClick();
 	}
 
-	void setEditor(JSONListCellEditor editor);
+	void setPresenter(Presenter editor);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/table/v2/results/cell/JSONListCellEditorView.java:8]: Rename only (and using interface).

File: src/main/java/org/sagebionetworks/web/client/widget/table/v2/results/cell/JSONListCellEditorViewImpl.java [hunk 3]
```diff
@@ -42,29 +38,22 @@
...
 	Widget widget;
-	JSONListCellEditor editor;
-
-	CellFactory cellFactory;
-	PortalGinInjector ginInjector;
+	Presenter presenter;
 
 	@Inject
-	public JSONListCellEditorViewImpl(Binder binder, CellFactory cellFactory) {
-		this.ginInjector = ginInjector;
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/table/v2/results/cell/JSONListCellEditorViewImpl.java:52]: Cleanup only.  Unused PortalGinInjector and CellFactory.

File: src/main/resources/org/sagebionetworks/web/client/widget/table/v2/results/cell/UserIdCellEditorViewImpl.ui.xml [hunk 4]
```diff
@@ -5,7 +5,7 @@
 	xmlns:bh="urn:import:org.gwtbootstrap3.client.ui.html">
 	<b:FormGroup ui:field="formGroup" addStyleNames="margin-bottom-0-imp">
 		<bh:Span ui:field="suggestBoxContainer" />
-		<g:FocusPanel addStyleNames="form-control cursor-text"
+		<g:FocusPanel addStyleNames="form-control cursor-text padding-2"
```
Comment (jay-hodgson) [inline src/main/resources/org/sagebionetworks/web/client/widget/table/v2/results/cell/UserIdCellEditorViewImpl.ui.xml:8]: Noticed single user editor has UserBadge that is outside textbox boundaries due to the padding.

**Review by zimingd — APPROVED** [review]

**Pull request merged**